pkg/controller/dnszone/dnszone_controller.go

```diff
@@ -0,0 +1,206 @@
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	controllerName     = "dnszone"
+	zoneResyncDuration = 2 * time.Hour
+)
+
+// Add creates a new DNSZone Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileDNSZone{
+		Client:           mgr.GetClient(),
+		scheme:           mgr.GetScheme(),
+		logger:           log.WithField("controller", controllerName),
+		awsClientBuilder: awsclient.NewClient,
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to DNSZone
+	err = c.Watch(&source.Kind{Type: &hivev1.DNSZone{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// watch DNSZone objects
+	err = c.Watch(&source.Kind{Type: &appsv1.Deployment{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &hivev1.DNSZone{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileDNSZone{}
+
+// ReconcileDNSZone reconciles a DNSZone object
+type ReconcileDNSZone struct {
+	client.Client
+	scheme *runtime.Scheme
+
+	logger log.FieldLogger
+
+	// awsClientBuilder is a function pointer to the function that builds the aws client.
+	awsClientBuilder func(kClient client.Client, secretName, namespace, region string) (awsclient.Client, error)
+}
+
+// Reconcile reads that state of the cluster for a DNSZone object and makes changes based on the state read
+// and what is in the DNSZone.Spec
+// Automatically generate RBAC rules to allow the Controller to read and write DNSZones
+// +kubebuilder:rbac:groups=hive.openshift.io,resources=dnszones,verbs=get;list;watch;create;update;patch;delete
+func (r *ReconcileDNSZone) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// Fetch the DNSZone object
+	desiredState := &hivev1.DNSZone{}
+	err := r.Get(context.TODO(), request.NamespacedName, desiredState)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Object not found, return.  Created objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers.
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+	dnsLog := r.logger.WithFields(log.Fields{
+		"DNSZone":   desiredState.Name,
```

Comment:
I might stick with name here for consistency, and we should add an identifier for the controller so we can filter to logs from specific controllers like we did in CO. I'll do the same in cluster deployment as soon as I'm in there again.